sonic-chassisd/tests/test_chassisd.py

```diff
@@ -366,3 +368,76 @@ def test_midplane_presence_supervisor():
     module_updater.deinit()
     fvs = midplane_table.get(name)
     assert fvs == None
+
+def test_asic_presence():
+    chassis = MockChassis()
+
+    #Supervisor
+    index = 0
+    name = "SUPERVISOR0"
+    desc = "Supervisor card"
+    slot = 16
+    module_type = ModuleBase.MODULE_TYPE_SUPERVISOR
+    supervisor = MockModule(index, name, desc, module_type, slot)
+    supervisor.set_midplane_ip()
+    chassis.module_list.append(supervisor)
+
+    #Linecard
+    index = 1
+    name = "LINE-CARD0"
+    desc = "36 port 400G card"
+    slot = 1
+    module_type = ModuleBase.MODULE_TYPE_LINE
+    module = MockModule(index, name, desc, module_type, slot)
+    module.set_midplane_ip()
+    chassis.module_list.append(module)
+
+    #Fabric-card with asics
+    index = 1
+    name = "FABRIC-CARD0"
+    desc = "Switch fabric card"
+    slot = 17
+    module_type = ModuleBase.MODULE_TYPE_FABRIC
+    fabric_asic_list = [("4", "0000:04:00.0"), ("5", "0000:05:00.0")]
+    fabric = MockModule(index, name, desc, module_type, slot, fabric_asic_list)
+    chassis.module_list.append(fabric)
+
+    #Run on supervisor
+    module_updater = ModuleUpdater(SYSLOG_IDENTIFIER, chassis)
+    module_updater.supervisor_slot = supervisor.get_slot()
+    module_updater.my_slot = supervisor.get_slot()
+    module_updater.modules_num_update()
+    module_updater.module_db_update()
+    module_updater.check_midplane_reachability()
+
+    #Asic presence on fabric module
+    fabric.set_oper_status(ModuleBase.MODULE_STATUS_ONLINE)
+    module_updater.module_db_update()
+    fabric_asic_table = module_updater.asic_table
+    assert len(fabric_asic_table.getKeys()) == 2
+
+    def test_fabric_asic(asic_name, asic_pci_address, module_name, asic_id_in_module):
```

Comment:
Seems this new test test_fabric_asic is not being run. To correct the function definition format here.
```
tests/test_chassis_db_init.py::test_provision_db PASSED                  [  8%]
tests/test_chassisd.py::test_moduleupdater_check_valid_fields PASSED     [ 16%]
tests/test_chassisd.py::test_moduleupdater_check_invalid_name PASSED     [ 25%]
tests/test_chassisd.py::test_moduleupdater_check_status_update PASSED    [ 33%]
tests/test_chassisd.py::test_moduleupdater_check_deinit PASSED           [ 41%]
tests/test_chassisd.py::test_configupdater_check_valid_names PASSED      [ 50%]
tests/test_chassisd.py::test_configupdater_check_valid_index PASSED      [ 58%]
tests/test_chassisd.py::test_configupdater_check_admin_state PASSED      [ 66%]
tests/test_chassisd.py::test_configupdater_check_num_modules PASSED      [ 75%]
tests/test_chassisd.py::test_midplane_presence_modules PASSED            [ 83%]
tests/test_chassisd.py::test_midplane_presence_supervisor PASSED         [ 91%]
tests/test_chassisd.py::test_asic_presence PASSED                        [100%]
```

Reply:
No, `test_fabric_asic` is not supposed to run as an unit test. 

It's part of `test_asic_presence`

`tests/test_chassisd.py::test_asic_presence PASSED                        [100%]`

Reply:
As a practice, we keep the function names starting with test_* as the testcases selected by pytest during the collection process. So we could remove the "test_" prefix for this function and define it outside the function def test_asic_presence ?

Reply:
Sure, I'll remove test_ prefix, but I think keeping it inside is clearer. Does it make sense to you?